geomstats/geometry/discrete_curves.py

```diff
@@ -1188,10 +1190,141 @@ class SRVMetric(PullbackDiffeoMetric):
+                "implemented for discrete curves embedded "
+                "in a Euclidean space."
+            )
+
+    def _define_embedding_space(self):
+        r"""Define embedding space with metric to pull back."""
+        space = self._space
+        discrete_curves = DiscreteCurves(
+            space.ambient_manifold, space.k_sampling_points, equip=False
+        )
+        discrete_curves.equip_with_metric(
+            SRVTranslationMetric, fiber_bundle=SRVTranslationBundle
+        )
+
+        factors = [space.ambient_manifold, discrete_curves]
+        embedding_space = ProductManifold(factors=factors, default_point_type="auto")
+        return embedding_space
+
+    def diffeomorphism(self, curves):
+        r"""Compute the image of a curve."""
+        curve_ndim = gs.ndim(curves)
+        curves = gs.to_ndarray(curves, to_ndim=3)
+        n_curves, k_sampling_points, dim = curves.shape
+
+        starting_point = curves[:, 0, :]
+        starting_points = copy.deepcopy(curves[:, 0, :])
+        starting_points = starting_points[:, None, :].repeat(k_sampling_points, axis=1)
+        curves_start_origin = curves - starting_points
+
+        result = gs.zeros((n_curves, k_sampling_points + 1, dim), dtype=float)
+
+        if curve_ndim == 2:
+            result[:, 0, :] = starting_point
+            result[:, 1:, :] = curves_start_origin
+            result = gs.squeeze(result)
+            result = result.flatten()
+            return result
+
+        result[:, 0, :] = starting_point
+        result[:, 1:, :] = curves_start_origin
+        result = gs.reshape(result, (n_curves, (k_sampling_points + 1) * dim))
+        return result
+
+    def inverse_diffeomorphism(self, image_point):
+        r"""Compute the curve of an image point."""
+        dim = self._space.ambient_manifold.dim
+        k_sampling_points = self._space.k_sampling_points
+
+        curve_ndim = gs.ndim(image_point)
+        image_points = gs.to_ndarray(image_point, to_ndim=2)
+        n_curves, _ = image_points.shape
+        starting_points = image_points[:, :dim]
+        curves_start_origin_flatten = image_points[:, dim:]
+        curves_start_origin = curves_start_origin_flatten.reshape(
+            (n_curves, k_sampling_points, dim)
+        )
+
+        starting_points = starting_points[:, None, :].repeat(k_sampling_points, axis=1)
+
+        curves = curves_start_origin + starting_points
+
+        if curve_ndim == 1:
+            return gs.squeeze(curves)
+
+        return curves
+
+    def tangent_diffeomorphism(self, tangent_vec, point):
+        r"""Equivalent to diffeomorphism."""
+        return self.diffeomorphism(tangent_vec)
+
+    def inverse_tangent_diffeomorphism(self, tangent_vec, image_point):
+        r"""Equivalent to inverse diffeomorphism."""
+        return self.inverse_diffeomorphism(tangent_vec)
+
+
+class SRVTranslationBundle(FiberBundle):
+    r"""Bundle of curves when you quotient by translation."""
```

Comment:
missing parameters of init here.

Reply:
And in the other python classes as well: please add them everywhere by taking example on other python classes of geomstats